### PR TITLE
Add the User table create script

### PR DESCRIPTION
**Pull request by Chris-Johnston**
Adds the script for creating the user table.

### CODE REVIEW

File: SQL/deleteme.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+delete after done making sql stuff
```
Comment (SaamIAm) [inline SQL/deleteme.txt]: You want delete this?

Comment (Chris-Johnston) [reply SQL/deleteme.txt]: Will delete in another branch.

**Review by SaamIAm — APPROVED** [review]
LGTM

File: SQL/Create_Table_User.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	[Username] [varchar](50) UNIQUE NOT NULL
+    -- Check that the username is valid
+    CONSTRAINT CHK_Username_Valid
+    -- Regex for usernames, any of these valid characters from 2-50 characters long
+    CHECK ([Username] LIKE '[^([!$?#\-_.0-9a-zA-Z]){2,50}$]'),
+    -- don't need to constrain password hashes, these are set by only the API level programatically
+    -- we can assume will always be valid
+	[PasswordHash] [binary](64) NOT NULL,
+    -- same goes for the HMAC key
+	[HMACKey] [binary](256) NOT NULL
```
Comment (anishprasad01) [inline SQL/Create_Table_User.sql:38]: Should there be ON DELETE and ON UPDATE statements for this table?

**Review by anishprasad01 — CHANGES_REQUESTED** [review]
ON UPDATE and ON DELETE statements in the create script

File: SQL/Create_Table_User.sql (hunk 2 above)
Comment (Chris-Johnston) [reply SQL/Create_Table_User.sql:38]: This can only be done for foreign keys, cannot be done table-wide. Invalid!

**Review by anishprasad01 — APPROVED** [review]

**Pull request merged**